Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/Reflection/LegacyReflectionBridge.cpp

```diff
@@ -270,137 +278,201 @@ namespace AZ::Reflection
                 m_serializeContext->EnumerateInstance(&context, nodeData.m_instance, nodeData.m_typeId, nullptr, nullptr);
             }
 
-            bool BeginNode(
-                void* instance, const AZ::SerializeContext::ClassData* classData, const AZ::SerializeContext::ClassElement* classElement)
+            void GenerateNodePath(const StackEntry& parentData, StackEntry& nodeData)
             {
-                // Ensure our instance pointer is resolved and safe to bind to member attributes
-                if (classElement)
-                {
-                    instance = AZ::Utils::ResolvePointer(instance, *classElement, *m_serializeContext);
-                }
-
-                StackEntry& parentData = m_stack.back();
                 AZStd::string path = parentData.m_path;
-                AZ::SerializeContext::IDataContainer::IAssociativeDataContainer* parentAssociativeInterface = nullptr;
                 if (parentData.m_classData && parentData.m_classData->m_container)
                 {
                     path.append("/");
                     path.append(AZStd::string::format("%zu", parentData.m_childElementIndex));
-
-                    auto& containerInfo = parentData.m_classData->m_container;
-                    parentAssociativeInterface = containerInfo->GetAssociativeContainerInterface();
                 }
-                else if (classElement)
+                else if (nodeData.m_classElement && nodeData.m_classElement->m_editData)
                 {
-                    AZStd::string_view elementName = classElement->m_name;
+                    AZStd::string_view elementName = nodeData.m_classElement->m_editData->m_name;
 
                     // Construct the serialized path for only those elements that have valid edit data. Otherwise, you can end up with
                     // serialized paths looking like "token1////token2/token3"
-                    if (!elementName.empty() && classElement->m_editData)
+                    if (!elementName.empty())
                     {
                         path.append("/");
                         path.append(elementName);
                     }
                 }
 
-                //! This is the case to create UIElements, Groups and Editor Data if any one of these is the first element in a class
-                auto iter = m_nonSerializedElements.begin();
-                while (iter != m_nonSerializedElements.end())
+                nodeData.m_path = AZStd::move(path);
+            }
+
+            bool IsInParentGroup(const StackEntry& nodeData, StackEntry& parentData)
```

Comment:
Nit: By this name I thought it was just checking if the node is in a parent group, but it's actually adding it to the group vector to be handled later, so maybe a different name like `NodeShouldBeHandledByParentGroup`. Not married to the verbiage, my suggestion is actually a little wordy, but something like that would be helpful.